packages/protocol-parser/src/decodeRecord.test.ts

```diff
@@ -21,4 +21,16 @@ describe("decodeRecord", () => {
       ]
     `);
   });
+
+  it("can decode an out of bounds array", () => {
+    const schema = { staticFields: [], dynamicFields: ["uint32[]"] } as const;
+    const values = decodeRecord(schema, "0x0000000000000000000000000000000000000000000000000400000000000004");
```

Comment:
The hex represents an encoded dynamic length field with total length 4 and length of the first array of 4, correct? Can you add more context for why the decoded array has one element with value 0? Somehow I would have expected either an empty because there is no data after the encoded data length, or an array with 4 elements that are all `0`?

Reply:
I think I had added this test for proof that `padSliceHex` was doing the right thing internally when used within `decodeRecord`. I can't recall where exactly I was getting a the hex value but I copied it from some tests or logs that were failing without `padSliceHex`.

To explain what's going on:

The total length and first array length is the byte length (4 bytes in the case of `uint32`). Meaning that there's one item/value for the `uint32[]` field.

The record hex is "trimmed" to the right-most byte (end of the counter). On chain, any bytes after the right-most byte would be treated as zeros, so we want to have the same behavior in the client.

So if we were to read the same value on chain, we should get a value of `[0]` for this record (one `uint32` value with unset bytes, i.e. `0`). This test ensures that the same behavior happens in the client.

Reply:
Ahh makes sense, had forgotten that the length corresponds to bytes and assumed elements. Thanks for the explanation!